rewrite-java/src/main/java/org/openrewrite/java/cleanup/UseSystemLineSeparator.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.cleanup;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesMethod;
+import org.openrewrite.java.tree.J;
+
+import java.time.Duration;
+import java.util.ArrayList;
+
+public class UseSystemLineSeparator extends Recipe {
+
+    private static final MethodMatcher GET_PROPERTY = new MethodMatcher("java.lang.System getProperty(..)");
+
+    @Override
+    public String getDisplayName() {
+        return "Use `System.lineSeparator()`";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Replaces calls to `System.getProperty(\"line.separator\")` with `System.lineSeparator()`.";
+    }
+    @Override
+    public Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(1);
+    }
+
+    @Override
+    protected @Nullable TreeVisitor<?, ExecutionContext> getSingleSourceApplicableTest() {
+        return new UsesMethod<>(GET_PROPERTY);
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new JavaIsoVisitor<ExecutionContext>() {
+            @Override
+            public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext context) {
+                J.MethodInvocation invocation = super.visitMethodInvocation(method, context);
+
+                if ("System.getProperty(\"line.separator\")".equalsIgnoreCase(invocation.toString())){
+
+                    return invocation
```

Comment:
It's better to do a `JavaTemplate` replacement here because that will also update the type attribution information on the method invocation.

Reply:
Is [this ](https://github.com/openrewrite/rewrite/pull/2778/commits/4bdca51236342a23ac8349398b115ef172263b1c) what you meant? 

Thank you!